### PR TITLE
Add checkstyle config

### LINKED ISSUE #10: Add Checkstyle config
Add Checkstyle config so that we can make sure all the codes following the same style

### PR DESCRIPTION
**Pull request by haocheng**
Motivation:

- Add Checkstyle config to make sure we are following the same style convention

Modifications:

- Add config in build.gradle to run checkstyle
- Add checkstyle rules configurations
- Fix checkstyle issues in the current code base
- Use `pull_request_target` to workaround Github Action permission issue

Result:

- Closes #10 
- All following PR will have checkstyle check

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -37,3 +39,10 @@ jobs:
         with:
           name: test-report
           path: build/reports/
+
+      - name: Upload checkstyle report
+        uses: actions/upload-artifact@v3
+        if: success() || failure()
+        with:
+          name: checkstyle-report
+          path: build/checkstyle/*.html
```
Comment (haocheng) [inline .github/workflows/test.yml:48]: I would like to display checkstyle report by using [Push Checkstyle Report](https://github.com/marketplace/actions/push-checkstyle-report) action but failed due to the `Error: HttpError: Resource not accessible by integration` [issue](https://github.com/jwgmeligmeyling/spotbugs-github-action/issues/30)...

File: build.gradle [hunk 2]
```diff
@@ -40,6 +41,15 @@ dependencies {
 	testImplementation "org.mockito.kotlin:mockito-kotlin:5.0.0"
 }
 
+def checkstyleConfigDir = rootProject.file('settings/checkstyle/')
+checkstyle {
+	configFile = new File(checkstyleConfigDir, 'checkstyle.xml')
+	configProperties = ['checkstyleConfigDir': "$checkstyleConfigDir"]
+	ignoreFailures = true
+	reportsDir = project.file('build/checkstyle')
+	toolVersion = '9.3'
```
Comment (haocheng) [inline build.gradle:50]: The latest version to support Java 8 is 9.x (10.x only supports Java 11)

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -17,12 +17,14 @@ jobs:
           java-version: ${{ matrix.java }}
           architecture: x64
       - uses: actions/checkout@v3
-    
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (haocheng) [inline .github/workflows/test.yml:21]: Used `${{ github.event.pull_request.head.sha }} ` according to the suggestion on [Push Checkstyle Report](https://github.com/marketplace/actions/push-checkstyle-report)
Screenshot 2023-07-04 at 9 39 06

File: settings/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -0,0 +1,350 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+  "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
```
Comment (haocheng) [inline settings/checkstyle/checkstyle.xml:6]: I have copied this file from [Armeria](https://github.com/line/armeria/) but removed checks related to Javadoc:

```
    <module name="JavadocType">
      <property name="scope" value="protected"/>
      <property name="excludeScope" value="package"/>
    </module>
    <module name="JavadocMethod">
      <property name="accessModifiers" value="protected, public"/>
      <property name="allowMissingParamTags" value="true"/>
      <property name="allowMissingReturnTag" value="true"/>
      <property name="allowedAnnotations" value="Override, Test"/>
    </module>
    <module name="MissingJavadocMethod">
      <property name="scope" value="protected"/>
    </module>
    <module name="MissingJavadocPackage"/>
    <module name="MissingJavadocType"/>
    <module name="JavadocParagraph"/>
    <module name="JavadocTagContinuationIndentation"/>
```

**Review by kojilin — APPROVED** [review]

**Pull request merged**